templates/rhpam74-authoring.yaml

```diff
@@ -836,7 +836,7 @@ objects:
             readOnly: true
 ## H2 volume mount BEGIN
           - name: "${APPLICATION_NAME}-h2-pvol"
-            mountPath: "/opt/eap/standalone/data"
+            mountPath: "/deployments/data/h2"
```

Comment:
Avoid to use /deployments, this directory is intended to contain only application files that will be deployed.
Instead use some directory on JBOSS_HOME, like /opt/eap/standalone/data/h2

Reply:
Hi! The `/opt/eap/standalone/data` isn't supposed to be used since it's locked for Artemis. That's one one the root causes of this problem.
The `/deployments/data` dir is the "default" directory per s2i docs. Perhaps `opt/eap/standalone/h2` or `opt/eap/standalone/db`? wdyt?

Reply:
Why is Artemis locking the entire /opt/eap/standalone/data/ directory? That directory is being mounted for more than one purpose...

In looking at rhpam73-authoring.yaml , I see a mix of directories at different levels, that we should probably cleanup...

rhpamcentr:
- volumeMount / name: "${APPLICATION_NAME}-rhpamcentr-pvol", mountPath: "/opt/eap/standalone/data/kie"
- env / name: GIT_HOOKS_DIR ; value: "${GIT_HOOKS_DIR}", example: /opt/eap/standalone/data/kie/git/hooks

kieserver:
- volumeMount / name: "${APPLICATION_NAME}-h2-pvol" , mountPath: "/opt/eap/standalone/data"
- env / name: RHPAM_XA_CONNECTION_PROPERTY_URL , value: "jdbc:h2:/opt/eap/standalone/data/rhpam"

Actually, looking again, perhaps only kieserver needs to change... How about this?:
- volumeMount / name: "${APPLICATION_NAME}-h2-pvol" , mountPath: "/opt/eap/standalone/data/kie"
- env / name: RHPAM_XA_CONNECTION_PROPERTY_URL , value: "jdbc:h2:/opt/eap/standalone/data/kie/h2/rhpam"

Reply:
@errantepiphany sounds good.

Reply:
1. We'd have to make sure we go through all the templates and make sure the paths are homogenized.
2. I don't remember which script(s) ensure those directories are created (`mkdir -p`), but we should make sure it's done.

Reply:
I'm not sure this will solve:
/opt/eap/standalone/data/kie/h2.

I need to investigate further, but I think won't work either because of the Artemis lock. This directory supposed to be used by it to write their journal. I'll investigate and report back.